src/main/java/Duke.java

```diff
@@ -1,10 +1,228 @@
+
 public class Duke {
-    public static void main(String[] args) {
+    private static void appendToFile(String filePath, String textToAppend) throws IOException {
+        FileWriter fw = new FileWriter(filePath, true); // create a FileWriter in append mode
+        fw.write(textToAppend);
+        fw.close();
+    }
+
+    private static void writeToFile(String filePath, String textToAdd) throws IOException {
+        FileWriter fw = new FileWriter(filePath);
+        fw.write(textToAdd);
+        fw.close();
+    }
+
+
+    public static void main(String[] args) throws Exception {
+
+        ArrayList<Task> tasks = new ArrayList<>();
+        //scan for duke.txt file and add as input first
+        int i = 0;
+
+        File duke=new File("Duke.txt");
+        if(duke.exists())
+        {
+            Scanner s = new Scanner(duke); // create a Scanner using the File as the source
+            while (s.hasNextLine()) {
+                String currentLine=s.nextLine();
+                Character taskLetter= currentLine.charAt(1);
+                Character tick=currentLine.charAt(4);
+
+
+                if(taskLetter=='T') {
+                    String taskToDo=currentLine.substring(7,currentLine.length());
+                    tasks.add(new Todo(taskToDo,2));
+                } else if (taskLetter=='D') {
+                    String[] ToSplitDeadline = currentLine.split("by:");
+
+                    String DeadlineTask = ToSplitDeadline[0].substring(7,ToSplitDeadline[0].length()-1);
+                    String date =ToSplitDeadline[1].substring(0,ToSplitDeadline[1].length()-1);
+                    tasks.add(new Deadline(DeadlineTask, date));
+
+                } else if(taskLetter=='E') {
+                    String[] ToSplitEvent = currentLine.split("from:");
+                    String EventTask = ToSplitEvent[0].substring(7,ToSplitEvent[0].length()-1);
+                    String[] date= ToSplitEvent[1].split("to:");
+                    tasks.add(new Event(EventTask,date[0], date[1]));
+                }
+                i+=1;
+
+                if(tick=='X') {
+                    tasks.get(i-1).markAsDone();
+                }
+
+
+            }
+        }
+
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+
+
+        String greeting = "____________________________________________________________\n"
+                + " Hello! I'm Duke\n"
+                + " What can I do for you?\n"
+                + "____________________________________________________________\n";
+
+        System.out.println(greeting);
+        Scanner in = new Scanner(System.in);
+        int index_for_mark= 1;
+
+        while (true) {
+
+            String line = in.nextLine();
+            //if bye
```

Comment:
Usually, you don't need to comment on things that may be seemingly obvious to the reader. So you comment minimally but sufficiently. [See here](https://nus-cs2113-ay2223s2.github.io/website/se-book-adapted/chapters/codeQuality.html#guideline-comment-minimally-but-sufficiently)